LCM/dsc/engine/ConfigurationManager/LocalConfigManagerHelper.c

```diff
@@ -1029,11 +1094,23 @@ MI_Result CallConsistencyEngine(
                 GetResourceString(ID_LCM_WRITEMESSAGE_ENDTESTPROCESSING_FALSE,&intlstr);   
                 ConcatStrings(&messageEvent,10,newLine, (MI_Char*)intlstr.str);
     
-                //Output event stating it resulted in false.
-                DSC_EventWriteMessageFromEngineConsistency(messageEvent);
+
+                
+                if (messageEvent != NULL)
+                {
+                    //Output event stating it resulted in false.
+                    DSC_EventWriteMessageFromEngineConsistency(messageEvent);
+                    DSC_free(messageEvent);
+                }
+
                 if(intlstr.str)
                     Intlstr_Free(intlstr);
-                result =  GetCimMIError(MI_RESULT_FAILED, cimErrorDetails, ID_LCM_WRITEMESSAGE_ENDTESTPROCESSING_FALSE);
+
+                // This is a difference between Windows DSC and Linux DSC in ApplyAndMonitor mode. Not sure why.
+                if (ShouldMonitor(configModeValue.string))
+                {
+                    result =  GetCimMIError(MI_RESULT_FAILED, cimErrorDetails, ID_LCM_WRITEMESSAGE_ENDTESTPROCESSING_FALSE);
+                }
```

Comment:
in monitor only mode looks like we fail engine operation when resource is not in desired state as test targetresource return false. Can you check if we don't fail on windows as not failing is the right behavior in this case for both MontiorOnly and Monitor . And then simply remove this GetCimError.

Reply:
I agree that the error shouldn't be there. It isn't on Windows. However, this could be considered a breaking change for ApplyAndMonitor mode on Linux since the error was already there. Are we ok making breaking changes?